CDP4Reporting/DataSource/ReportingDataSourceNode.cs

```diff
@@ -0,0 +1,288 @@
+        /// </summary>
+        internal List<ReportingDataSourceNode<T>> Children { get; } = new List<ReportingDataSourceNode<T>>();
+
+        /// <summary>
+        /// The <see cref="CDP4Common.EngineeringModelData.NestedElement"/> associated with this node.
+        /// </summary>
+        internal readonly NestedElement NestedElement;
+
+        /// <summary>
+        /// The <see cref="CDP4Common.EngineeringModelData.ElementBase"/> associated with this node.
+        /// </summary>
+        internal ElementBase ElementBase =>
+            this.NestedElement.GetElementBase();
+
+        /// <summary>
+        /// The <see cref="CDP4Common.EngineeringModelData.ElementDefinition"/> representing this node.
+        /// </summary>
+        internal ElementDefinition ElementDefinition =>
+            this.NestedElement.GetElementDefinition();
+
+        /// <summary>
+        /// The <see cref="ElementUsage"/> representing this node, if it exists.
+        /// </summary>
+        internal ElementUsage ElementUsage =>
+            this.NestedElement.GetElementUsage();
+
+        /// <summary>
+        /// The filtering <see cref="CategoryHierarchy"/> that must be matched on the current <see cref="ElementBase"/>.
+        /// </summary>
+        private readonly CategoryHierarchy filterCategory;
+
+        /// <summary>
+        /// Gets the name of the field/column when the data is transfered to a <see cref="DataRow"/>.
+        /// </summary>
+        private string fieldName => this.filterCategory?.FieldName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReportingDataSourceNode{T}"/> class.
+        /// </summary>
+        /// <param name="categoryHierarchy">
+        /// The <see cref="CategoryHierarchy"/> associated with this node's subtree.
+        /// </param>
+        /// <param name="topElement">
+        /// The <see cref="CDP4Common.EngineeringModelData.NestedElement"/> associated with this node.
+        /// </param>
+        /// <param name="parent">
+        /// The parent node in the hierarhical tree upon which the data source is based.
+        /// </param>
+        public ReportingDataSourceNode(
+            CategoryHierarchy categoryHierarchy,
+            NestedElement topElement,
+            ReportingDataSourceNode<T> parent = null)
+        {
+            this.filterCategory = categoryHierarchy;
+            this.NestedElement = topElement;
+            this.parent = parent;
+            this.rowRepresentation = this.GetRowRepresentation();
+        }
+
+        /// <summary>
+        /// Checks if a category was found in this instance or up its parent tree.
+        /// </summary>
+        /// <param name="category">
+        /// The <see cref="Category"/>.
+        /// </param>
+        /// <returns>
+        /// True if found, otherwise false.
+        /// </returns>
+        internal bool HadCategoryUpTree(Category category)
```

Comment:
Why is this method using the past tense? The convention seems to be to use the present.